test/interactions/keyInteractionTests.ts

```diff
@@ -4,91 +4,62 @@ var assert = chai.assert;
 
 describe("Interactions", () => {
   describe("KeyInteraction", () => {
-    it("Triggers appropriate callback for the key pressed", () => {
-      var svg = TestMethods.generateSVG(400, 400);
-      var component = new Plottable.Component();
+    var svg: d3.Selection<void>;
+    var component: Plottable.Component;
+    var keyInteraction: Plottable.Interactions.Key;
+    var aCode: number;
+    var bCode: number;
+
+    var aCallbackCalled: boolean;
+    var bCallbackCalled: boolean;
+    var aCallback: (asd: number) => void;
+    var bCallback: (asd: number) => void;
+    var $target: JQuery;
+
+    beforeEach(() => {
+      aCode = 65;
+      bCode = 66;
+      component = new Plottable.Component();
+      svg = TestMethods.generateSVG(400, 400);
+      keyInteraction = new Plottable.Interactions.Key();
+      aCallback = () => aCallbackCalled = true;
+      bCallback = () => bCallbackCalled = true;
       component.renderTo(svg);
-
-      var keyInteraction = new Plottable.Interactions.Key();
-
-      var aCode = 65; // "a" key
-      var bCode = 66; // "b" key
-
-      var aCallbackCalled = false;
-      var aCallback = () => aCallbackCalled = true;
-      var bCallbackCalled = false;
-      var bCallback = () => bCallbackCalled = true;
-
-      keyInteraction.onKeyPress(aCode, aCallback);
-      keyInteraction.onKeyPress(bCode, bCallback);
-      keyInteraction.attachTo(component);
-
-      var $target = $(component.background().node());
-
-      TestMethods.triggerFakeMouseEvent("mouseover", component.background(), 100, 100);
-      $target.simulate("keydown", { keyCode: aCode });
-      assert.isTrue(aCallbackCalled, "callback for \"a\" was called when \"a\" key was pressed");
-      assert.isFalse(bCallbackCalled, "callback for \"b\" was not called when \"a\" key was pressed");
-
-      aCallbackCalled = false;
-      $target.simulate("keydown", { keyCode: bCode });
-      assert.isFalse(aCallbackCalled, "callback for \"a\" was not called when \"b\" key was pressed");
-      assert.isTrue(bCallbackCalled, "callback for \"b\" was called when \"b\" key was pressed");
-
-      TestMethods.triggerFakeMouseEvent("mouseout", component.background(), -100, -100);
+      $target = $(component.background().node());
```

Comment:
This sounds over specific for a beforeEach(). I would just duplicate and inline this bit in all test.